bsdlib/doc/bsdlib_ug_adapting.rst

```diff
@@ -0,0 +1,360 @@
+   * Report back the remaining time of the timer if the specific timer is interrupted.
+   * If timed out, report NRF_ETIMEDOUT.
+
+3. bsd_os_errno_set
+
+   This function translates errno’s from the BSD library to the OS-defined ones.
+
+   *Required action*:
+
+   * Implement a translation for each errno set by the BSD library.
+     If it overlaps with your OS errno, translation is not needed.
+
+4. bsd_os_application_irq_clear
+
+   This function is called by the BSD library when the library wants to clear IRQ on the low priority BSD library scheduling IRQ.
+
+   *Required action*:
+
+   * Clear the low priority BSD library scheduling IRQ using OS primitives or NVIC functions.
+
+5. bsd_os_application_irq_set
+
+   This function is called by the BSD library when the library wants to set a pending IRQ on the low priority BSD library scheduling IRQ.
+
+   *Required action*:
+
+   * Set a pending IRQ on the low priority BSD library scheduling IRQ using OS primitives or NVIC functions.
+
+6. bsd_os_trace_irq_clear
+
+   This function is called by the BSD library when the library wants to clear IRQ on the low priority trace scheduling IRQ.
+
+   *Required action*:
+
+   * Clear the low priority trace scheduling IRQ using OS primitives or NVIC functions.
+
+7. bsd_os_trace_irq_set
+
+   This function is called by the BSD library when the library wants to set a pending IRQ on the low priority trace scheduling IRQ.
+
+   *Required action*:
+
+   * Set a pending IRQ on the low priority trace scheduling IRQ using OS primitives or NVIC functions.
+
+8. bsd_os_trace_put
+
+   This function puts the trace string to the desired medium, typically UART.
+   However, the medium used to forward and store the traces is up to the implementation and must be initialized correctly before used.
+   If you are not interested in traces, they can be ignored and this function can be empty and simply return.
+
+9. bsd_os_application_irq_handler
+
+   This function is implemented in the BSD library and must be called upon the low priority BSD library IRQ handler, triggered by the :cpp:func:`bsd_os_application_irq_set` function.
+
+10.	bsd_os_trace_irq_handler
+
+    This function is implemented in the BSD library and must be called upon the low priority trace IRQ handler, triggered by the :cpp:func:`bsd_os_trace_irq_set` function.
+
+11.	Other scenarios to handle in :file:`bsd_os.c`
+
+    In case the OS has its own IRQ handler scheme, not directly forwarding the IPC_IRQHandler to the BSD library, this must be routed by the OS.
+	The OS must call IPC_IRQHandler() upon all IRQs with IRQ number IPC_IRQn.
+
+	In :file:`bsd_os.c`, you can also configure a desired medium for forwarding trace data upon :cpp:func:`bsd_os_trace_put` calls .
+
+Memory
+******
+
+The BSD library uses fixed address at an absolute position in RAM for communication with the modem.
+The reserved memory slot is 64 Kb in size.
```

Comment:
We could address the start/end addresses in text directly.

Reply:
just wanted to remark that you introduced a define for this thats different to 64 Kb.

in bsd_platform.h:
#define BSD_RESERVED_MEMORY_SIZE               0xc020

Reply:
Thanks @tothero , it's a very good point! 

I'll propose a new text here. The truth is 0xc020. The 64Kb is just for convenience to have a round number. I'll try to give some feedback to the RAM partition image offline to @b-gent, to highlight that we use seven 8Kb RAM blocks configured non-secure in the lower 128Kb region. The spare 8160 bytes in the 7th block is really free for non-secure application usage.